### PR TITLE
Update axtls

### PR DESCRIPTION
**Pull request by kealist**
This is a rough update to the [axtls](http://axtls.sourceforge.net/index.htm) portions of the crypt library.  

Needs some work, still a WIP, but thought it would be good to compare anything that has changed.  Some pieces were already modified for rebol usage, and I tried to keep those intact.  @hostilefork Wanted to clean up trouble spots before I get too far.

This does not extend Rebol's capability at all.

### CODE REVIEW

**Comment by kealist** [comment]
Seems a couple unused parameters that didn't show up on my build

**Comment by hostilefork** [comment]
Hmmm...doesn't look like a whole lot has changed (!)

If it works, I don't see any real problem with patching to use code that's closer to the original...and also important, would be keeping links to where any original code came from.

But as people have mentioned, it's kind of a grab bag of things.  Incorporating libSodium has come up a lot.  I'm not sure what kind of size difference we'd be looking at if that were used instead, though.

In short, no super-strong opinions on my end, just as long as any updates don't break things.  :-)

File: src/extensions/crypt/rsa/rsa.c [hunk 1]
```diff
@@ -193,21 +189,26 @@ void RSA_free(RSA_CTX *rsa_ctx)
...
+ * @param out_data [out] The decrypted data.
+ * @param out_len [int] The size of the decrypted buffer in bytes
  * @param is_decryption [in] Decryption or verify operation.
  * @return  The number of bytes that were originally encrypted. -1 on error.
  * @see http://www.rsasecurity.com/rsalabs/node.asp?id=2125
  */
-int RSA_decrypt(const RSA_CTX *ctx, const uint8_t *in_data,
-                            uint8_t *out_data, int is_decryption, int padding)
+int RSA_decrypt(const RSA_CTX *ctx, const uint8_t *in_data, 
+                            uint8_t *out_data, int out_len, int is_decryption)
```
Comment (hostilefork) [inline src/extensions/crypt/rsa/rsa.c]: When they dropped the padding from the `RSA_encrypt`, you noticed and could drop the parameter.  When they dropped the padding from `RSA_decrypt`, they made a completely different integer parameter in the same spot.

When I think about the idea of these kinds of codebases and "security" it reminds me of The Princess Bride.  *"You keep using that word.  I don't think it means what you think it means."*  :-/

Comment (kealist) [reply src/extensions/crypt/rsa/rsa.c]: Yes, that is terrible.

File: src/extensions/crypt/aes/aes.c [hunk 2]
```diff
@@ -200,7 +200,6 @@ void AES_set_key(AES_CTX *ctx, const uint8_t *key,
             return;
     }
 
-    ctx->key_mode = AES_MODE_ENCRYPT; //default mode
```
Comment (hostilefork) [inline src/extensions/crypt/aes/aes.c]: via plain 'ol process of elimination, it seems that beyond the other thing, this `ctx->key_mode` setting (here and for AES_MODE_DECRYPT) appears needed to make `read https://example.com` work.

Go figure.

File: src/extensions/crypt/rsa/rsa.c (hunk 1 above)
Comment (hostilefork) [reply src/extensions/crypt/rsa/rsa.c]: Mathematician <> Programmer

**Comment by hostilefork** [comment]
In the interest of tying up loose ends, fixed up 'n' merged:

https://travis-ci.org/metaeducation/ren-c/builds/369145112

...and if we're all feeling in the tying-up-loose-end kind of mood, the situation with the serial is more approachable now.  There is no more REBCHR *(well, the name was retaken for something else, but the concept of there being string abstractions that are maybe-wide, or maybe-not, is gone)*.

Serial now is a client of the libRebol API, and along with other "Rebdev" based bits a candidate for being turned into an extension.  But debugging should hopefully be easier, and feedback more relevant:

https://github.com/metaeducation/ren-c/blob/master/src/core/p-serial.c
https://github.com/metaeducation/ren-c/blob/master/src/os/windows/dev-serial.c
https://github.com/metaeducation/ren-c/blob/master/src/os/posix/dev-serial.c

Feedback on the API is relevant.  And the thing about having these examples, even if the REBREQ or whatever is crufty, is that it gives good working cases to study when looking at what a "real" good PORT! design would look like.

So it would be a good time to wire up some serial things and see what's up...

**Comment by kealist** [comment]
Thank you for fixing that up.   I will try to look at serial again now.  Having been doing embedded C work for the last 5 months, I probably can manage it better anyways
